libs/user-facing-error-parsing/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+extern crate proc_macro;
+use syn::{Attribute, ItemStruct};
+
+/* This parses the list of files, looks for all structs and generates a vector of error types:
```

Comment:
Comments like this are preferred in the /// style, then they also show up for hover tooltips.